### PR TITLE
[Choreo] Add support to send oauth2 token in sec-websocket-protocol header

### PR DESCRIPTION
**Pull request by Thushani-Jayasekera**
### Issues

Fixes #

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java [hunk 1]
```diff
@@ -102,9 +104,19 @@ public JWTAuthenticator() {
...
-            return jwt != null && jwt.split("\\.").length == 3;
+
+            if (apiType.equalsIgnoreCase(APIConstants.ApiType.WEB_SOCKET)) {
+                if (jwt == null) {
+                    jwt = extractJWTInWSProtocolHeader(requestContext);
+                }
+                AuthenticatorUtils.addWSProtocolResponseHeaderIfRequired(requestContext,
+                        Constants.WS_OAUTH2_KEY_IDENTIFIED);
+
+                return jwt != null && jwt.split("\\.").length == 3;
```
Comment (renuka-fernando) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java:118]: shouldn't this return be outside of the if block?

Comment (Thushani-Jayasekera) [reply enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java]: yes

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/AuthenticatorUtils.java [hunk 2]
```diff
@@ -41,4 +46,24 @@ public static void populateTierInfo(APIKeyValidationInfoDTO validationInfo, JWTC
             }
         }
     }
+
+    public static void addWSProtocolResponseHeaderIfRequired(RequestContext requestContext, String protocolKeyword) {
+        String secProtocolHeader = requestContext.getHeaders().get(HttpConstants.WEBSOCKET_PROTOCOL_HEADER);
+        if (secProtocolHeader != null) {
+            String[] secProtocolHeaderValues = secProtocolHeader.split(",");
+            if (secProtocolHeaderValues[0].equals(protocolKeyword) &&
```
Comment (renuka-fernando) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/AuthenticatorUtils.java:54]: shouldn't we check the length of the array before reading 0th index?

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java [hunk 3]
```diff
@@ -766,4 +778,17 @@ private String getJWTTokenIdentifier(SignedJWTInfo signedJWTInfo) {
         }
         return signedJWTInfo.getSignedJWT().getSignature().toString();
     }
+
+    public String extractJWTInWSProtocolHeader(RequestContext requestContext) {
+        String protocolHeader = requestContext.getHeaders().get(
+                HttpConstants.WEBSOCKET_PROTOCOL_HEADER);
+        if (protocolHeader != null) {
+            String[] secProtocolHeaderValues = protocolHeader.split(",");
+            if (secProtocolHeaderValues.length > 1 && secProtocolHeaderValues[0].equals(
```
Comment (renuka-fernando) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java:787]: What if WEBSOCKET_PROTOCOL_HEADER contains `chat, choreo-token, <TOKEN>`?

Comment (Thushani-Jayasekera) [reply enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java:787]: The user is supposed to send in the given order. First token information and then backend protocol information. 

We have implemented in a similar way for the handling of API key and the console sends the information in this format.

**Comment by choreo-cicd** [comment]
[succeeded]  : dev-deployment-v2 : 20240927.19

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20240927.19

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : stage-deployment-v2 : 20240930.1

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20240930.1

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20240930.1

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20240930.1

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20240930.1

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : stage-deployment-v2 : 20240930.1

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20240930.2

**Comment by choreo-cicd** [comment]
[] Dataplane(EastUS) cluster : stage-deployment-v2 : 20240930.2

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : stage-deployment-v2 : 20240930.2

**Comment by choreo-cicd** [comment]
[]  : dev-deployment-v2 : 20240930.17

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20241001.2

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20241001.2

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(CDP2-WU2) cluster : prod-deployment-v2 : 20241001.2

**Pull request merged**